docstore/internal/fields/fields.go

```diff
@@ -178,16 +178,32 @@ func (c *Cache) Fields(t reflect.Type) (List, error) {
 // A List is a list of Fields.
 type List []Field
 
-// Match returns the field in the list whose name best matches the supplied
+// MatchExact returns the field in the list with the given name, or nil if there is
```

Comment:
My only concern is that we should document when to use which "match" somewhere, maybe here in the comment is fine.

Reply:
What do you mean by "when to use which 'match'"? Can you give me an example?

Reply:
I meant there are `MatchExact` and `MatchFold`, we are using `MatchFold` at the moment, it would be helpful to document why we need both for future reference.

Reply:
Oh, I think of this as a general-purpose package. Someday I want to move it out of here, maybe to its own repo. For now, I don't want to expose it to the world so it lives in internal. But it isn't specific to docstore.

So I think that comment is a good idea, but instead of putting it here I'd put it in driver/codec.go, where we call the match function. You can do that as part of #2415.